### PR TITLE
logi page + store + service layer + sagas

### PR DESCRIPTION
**Pull request by CherniyIrtish**

### CODE REVIEW

File: src/app/core/http-services/auth.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const axios = require('axios');
+
+
+export class Auth {
+
+  public async getToken(loginForm: {[key: string]: any}): Promise<any> {
+    const response = await axios.post('http://localhost:4000/api/auth', loginForm);
```
Comment (CherniyIrtish) [inline src/app/core/http-services/auth.ts:7]: move to API variable

File: package.json [hunk 2]
```diff
@@ -39,5 +44,8 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "devDependencies": {
+    "@types/react-redux": "^7.1.9"
```
Comment (CherniyIrtish) [inline package.json:49]: remove bird

File: src/app/core/http-services/auth.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const axios = require('axios');
+
+
+export class Auth {
+
+  public async getToken(loginForm: {[key: string]: any}): Promise<any> {
```
Comment (CherniyIrtish) [inline src/app/core/http-services/auth.ts:6]: add interfaces

File: src/app/store/actions/index.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export {}
```
Comment (CherniyIrtish) [inline src/app/store/actions/index.ts:1]: add combine export

File: src/app/store/actions/token-actions.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { GET_TOKEN } from '../types';
+
+export const getToken = (formValue: {email: string, password: string}) => {
```
Comment (CherniyIrtish) [inline src/app/store/actions/token-actions.ts:3]: do we need SET_TOKEN action? should be checked

File: src/app/store/sagas/token-sagas.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  yield takeEvery(GET_TOKEN, getToken);
+}
+
+function* getToken(action: { type: string, payload: { [key: string]: any } }) {
+  try {
+    const dataResponse = yield authService.getToken(action.payload);
+
+    yield put({type: SET_TOKEN, token: dataResponse.token})
+  } catch (error) {
+    console.log('error ', error)
```
Comment (CherniyIrtish) [inline src/app/store/sagas/token-sagas.ts:17]: add task for error handling

File: src/app/store/types.ts [hunk 7]
```diff
@@ -0,0 +1,2 @@
+export const GET_TOKEN = 'GET_TOKEN';
```
Comment (CherniyIrtish) [inline src/app/store/types.ts:1]: divide types to inner and outer

File: package.json (hunk 2 above)
Comment (IgorMitropan) [reply package.json:49]: bird)))

File: src/app/store/actions/token-actions.ts (hunk 5 above)
Comment (IgorMitropan) [reply src/app/store/actions/token-actions.ts:3]: We usually call such actions with SUCCESS postfix, e,g, GET_TOOKEN_SUCCESS and for sure we need this action.

File: src/app/store/sagas/token-sagas.ts (hunk 6 above)
Comment (IgorMitropan) [reply src/app/store/sagas/token-sagas.ts:17]: Here we should PUT (dispatch) GET_TOKEN_ERROR action.

File: src/app/store/store.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+import rootReducer from './reducers';
+import createSagaMiddleware from 'redux-saga';
+import rootSaga from './sagas';
+
+const sagaMiddleware = createSagaMiddleware();
+
+// @ts-ignore
+const store = createStore(rootReducer, compose(applyMiddleware(sagaMiddleware), window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()));
+
+sagaMiddleware.run(rootSaga);
```
Comment (IgorMitropan) [inline src/app/store/store.ts:11]: I suggest using redux-ducks structure from the very beginning.

Comment (CherniyIrtish) [reply src/app/store/store.ts:11]: what do you mean?

**Pull request merged**